policy/nomad/source.go

```diff
@@ -94,12 +94,6 @@ func (s *Source) MonitorIDs(ctx context.Context, resultCh chan<- []policy.Policy
 				continue
 			}
 
-			// If the index has not changed, the query returned because the timeout
-			// was reached, therefore start the next query loop.
-			if !blocking.IndexHasChanged(meta.LastIndex, q.WaitIndex) {
```

Comment:
> the policy list ID monitor would silently continue if the Autoscler re-connected to a Nomad cluster where the list of policies were different but the modify index had the same value.

I guess this removal relates to the above PR mention. If not please ignore this comment. In what situation would we reconnect to a cluster that has an altered list output but the table index has not changed?

Reply:
It's an odd case that is very unlikely to happen in production, but if you run a Nomad agent in dev mode, register a job with a scaling policy, then stop the dev agent and register the same job again, you will end up with the same index (since the same modifications were executed) but a different list (the policy ID will be different).

The root of this issue is that lists don't have an ID, so you can't really match an index to a list like, for example, a policy. The only way to make sure we have the same list is by iterating over the policy IDs, which we would anyway in the policy handler.

_But_ this is a very unusual case that is extremely unlikely to happen in a real environment. Also, this was an early "fix". Re-running the policy manager on disconnect actually re-creates this monitoring goroutine, so this problem doesn't happen anymore.

I will add this check back since it's useful to avoid unnecessary work.